libs/common/src/change-stream/test-helpers/with-deletion-audit-logs.api-tests.ts

```diff
@@ -0,0 +1,192 @@
+import { Collection, ObjectId, WithoutId } from 'mongodb';
+import { when, WhenMock } from 'jest-when';
+import { MongoDbClient } from '../../mongo-db-client';
+import { AuditDatabaseRecord, DeletionAuditLog, MongoDbCollectionName } from '../../types';
+
+type Params = {
+  makeRequest: () => Promise<void>;
+  collectionName: MongoDbCollectionName;
+  auditRecord: AuditDatabaseRecord;
+  getDeletedDocumentId: () => ObjectId;
+  mockGetCollection: jest.Mock;
+};
+
+export const withDeletionAuditLogsTests = ({ makeRequest, collectionName, auditRecord, getDeletedDocumentId, mockGetCollection }: Params) => {
+  describe(`when deleting a document from ${collectionName}`, () => {
+    let mongoDbClient: MongoDbClient;
+    let deletionAuditLogsCollection: Collection<WithoutId<DeletionAuditLog>>;
+
+    beforeAll(async () => {
+      mongoDbClient = new MongoDbClient({
```

Comment:
Could instantiating another client cause race conditions? Could pass in the real client?